File: src/com/leozhi/topic242/Solution03.java
package com.leozhi.topic242;

/**
 * @author leozhi
 * 哈希表
 * 官方题解
 * 通过
 * 5ms
 */
public class Solution03 {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }
        int[] table = new int[26];
        for (int i = 0; i < s.length(); i++) {
            table[s.charAt(i) - 'a']++;
        }
        for (int i = 0; i < t.length(); i++) {
            if (--table[t.charAt(i) - 'a'] < 0) {
                return false;
            }
        }
        return true;
    }
}

File: src/com/leozhi/topic976/Solution03.kt
package com.leozhi.topic976

/**
 * @author leozhi
 * 贪心算法
 * 通过
 * 292ms
 */
class Solution03 {
    fun largestPerimeter(A: IntArray): Int {
        A.sort()
        for (i in A.size - 1 downTo 2) {
            if (A[i - 2] + A[i - 1] > A[i]) {
                return A[i - 2] + A[i - 1] + A[i]
            }
        }
        return 0
    }
}